af_cv.cpp

```diff
@@ -166,21 +166,25 @@ void array_to_mat(const array& input_, cv::Mat& output, int type, bool transpose
     } else {
         input = input_;
     }
-    output = cv::Mat(input.dims(ndims - 1), input.dims(ndims - 2), CV_MAKETYPE(type, channels));
+    if(ndims == 1) {
+        output = cv::Mat(input.dims(ndims - 1), input.elements(), CV_MAKETYPE(type, channels));
+    } else {
+        output = cv::Mat(input.dims(ndims - 1), input.dims(ndims - 2), CV_MAKETYPE(type, channels));
+    }
     if (type == CV_32F) {
         float* data = output.ptr<float>(0);
         input.host((void*)data);
     } else if (type == CV_32S) {
         int* data = output.ptr<int>(0);
-        input.as(af::s32).host((void*)data);
+        input.as(s32).host((void*)data);
```

Comment:
if af:: no longer needed?

Reply:
ah, i see 'using af' is in header, oops
up to you if you prefer to remove af:: or keep

Reply:
actually ArrayFire data types aren't directly under af::, they are under af::dtype::
```suggestion
        input.as(af::dtype::s32).host((void*)data);
```